packages/mobile/src/hooks/chat/useChatSettings.ts

```diff
@@ -0,0 +1,26 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { useEffect, useState } from 'react';
+
+const ALLOW_YOUTUBE_PLAYBACK = 'youtubeplayback';
+
+export const useChatSettings = () => {
+    const [allowYoutubePlayback, setAllowYoutubePlayback] = useState<boolean>(true);
+
+    // Fetching default value
+    useEffect(() => {
+        (async () => {
+            const value = await AsyncStorage.getItem(ALLOW_YOUTUBE_PLAYBACK);
+            setAllowYoutubePlayback(value === 'true');
+        })();
+    }, []);
+
+    // When state changes, save to AsyncStorage
+    useEffect(() => {
```

Comment:
Probably better to have the exported `setAllowYoutubePlayback` to set the state and then update the asyncStorage. You usually wanna avoid useEffect as much as possible, especially if it's such an async/await mess, then we always wanna avoid it as much as possible